go/enclave/evm/evm_facade.go

```diff
@@ -104,23 +107,63 @@ func (exec *evmExecutor) execute(tx *common.L2PricedTransaction, from gethcommon
 		return nil, err
 	}
 
+	actualGasLimit := msg.GasLimit // before l1 cost is applied
+
 	receipt, err := adjustPublishingCostGas(tx, msg, s, header, noBaseFee, func() (*types.Receipt, error) {
 		s.SetTxContext(tx.Tx.Hash(), tCount)
 		return gethcore.ApplyTransactionWithEVM(msg, gp, s, header.Number, header.Hash(), header.Time, tx.Tx, usedGas, evmEnv)
 	})
 	if err != nil {
 		return nil, err
 	}
+	if receipt == nil {
+		return nil, fmt.Errorf("internal: nil receipt returned from execution")
+	}
 	receipt.Logs = s.GetLogs(tx.Tx.Hash(), header.Number.Uint64(), header.Hash(), header.Time)
 
 	contractsWithVisibility := make(map[gethcommon.Address]*core.ContractVisibilityConfig)
+
+	// Compute leftover user gas after the main execution
+	var gasLeft uint64 = actualGasLimit
+	if receipt.GasUsed > gasLeft {
```

Comment:
Let's write this as: `if receipt.GasUsed > actualGasLimit`. (It's very confusing to compare with `gasLeft`)

This should be much easier to follow:
```go
if receipt.GasUsed > actualGasLimit {
  return error..
}
gasLeft:=actualGasLimit - receipt.GasUsed
```